staging/src/k8s.io/apiserver/pkg/storage/etcd3/event.go

```diff
@@ -53,5 +54,9 @@ func parseEvent(e *clientv3.Event) *event {
 	if e.PrevKv != nil {
 		ret.prevValue = e.PrevKv.Value
 	}
-	return ret
+	if ret.isDeleted && ret.prevValue == nil {
```

Comment:
if the previous value existed, but was a zero-length byte array, does this show up as nil here? do we know if `e.PrevKv` as nil as well when we encountered this?

Reply:
> if the previous value existed, but was a zero-length byte array, does this show up as nil here?

I expect it would show up as the zero length byte array

>  do we know if e.PrevKv as nil as well when we encountered this?

All I can be certain is that when I encountered this either e.PrevKv was nil or e.PrevKv.Value was nil. I expect that e.PrevKv was nil though, given the info I had on compaction taking place and [this comment](https://sourcegraph.com/github.com/kubernetes/kubernetes@2347da833c3d9e956b30028c1b7e0e891d19d44d/-/blob/vendor/github.com/coreos/etcd/clientv3/op.go#L461)

Reply:
> I expect it would show up as the zero length byte array

expectations are funny things... if it uses protobuf, zero-length and nil arrays [can be serialized identically](https://github.com/kubernetes/kubernetes/issues/43203)

Reply:
Ok, I was mostly trying to fix the specific problem I encountered, to error here if and only if the prev value was nil, and not erroring on the empty array as I'm not sure we have that guarantee here.

Do you think this should error for both? error if `len(ret.preValue)==0`?

Reply:
if e.PrevKv is nil, that seems like a much clearer indicator of a problem